nodeadm/internal/system/environment.go

```diff
@@ -93,6 +118,43 @@ func (a *environmentAspect) generateSystemdConfig(envVars map[string]string) str
 	return builder.String()
 }
 
+func (a *environmentAspect) writeServiceDropinConfig(serviceName, dropinPath string, envVars map[string]string) error {
+	if len(envVars) == 0 {
+		return nil
+	}
+
+	log := zap.L()
+	log.Info("Writing environment variables to service drop-in",
+		zap.String("service", serviceName),
+		zap.String("path", dropinPath),
+		zap.Int("count", len(envVars)))
+
+	content := a.generateServiceDropinConfig(envVars)
+
+	if err := util.WriteFileWithDir(dropinPath, []byte(content), 0644); err != nil {
+		return fmt.Errorf("failed to write service drop-in config: %w", err)
+	}
+
+	log.Info("Reloading systemd configuration for service drop-in", zap.String("service", serviceName))
```

Comment:
Do you need to reload config for every file you write? Or can you just reload once at the end after writing all the config files? (Maybe making sure not to reload if it's not needed)